eating_cookies/eating_cookies.py

```diff
@@ -5,12 +5,20 @@
 # The cache parameter is here for if you want to implement
 # a solution that is more efficient than the naive 
 # recursive solution
-def eating_cookies(n, cache=None):
-  pass
+def eating_cookies(n, cache={}):
+  if n == 0:
+    return 1
+  if n < 0:
+    return 0
+  if n not in cache:
+    value = eating_cookies(n-1) + eating_cookies(n-2) + eating_cookies(n-3)
```

Comment:
Well done finding this pattern as well as building up a cache to improve the efficiency of the solution.